src/Raven.Studio/typescript/common/shell/databasesManager.ts

```diff
@@ -62,22 +63,29 @@ class databasesManager {
             });
     }
 
-    activateBasedOnCurrentUrl(): JQueryPromise<canActivateResultDto> | boolean {
-        const dbUrl = appUrl.getDatabaseNameFromUrl();
-        if (dbUrl) {
-            const db = this.getDatabaseByName(dbUrl);
-            return this.activateIfDifferent(db, dbUrl);
+    activateBasedOnCurrentUrl(dbName: string, isViewAllowed: boolean): JQueryPromise<canActivateResultDto> | boolean {
+        if (dbName) {
+            const db = this.getDatabaseByName(dbName);
+            return this.activateIfDifferent(db, dbName, isViewAllowed);
         }
         return true;
     }
 
-    private activateIfDifferent(db: database, dbName: string): JQueryPromise<canActivateResultDto> {
+    private activateIfDifferent(db: database, dbName: string, isViewAllowed: boolean): JQueryPromise<canActivateResultDto> {
```

Comment:
can we check if view is allowed inside this method, instead of passing this around?

Reply:
Don't we want the check to be made after `this.initialized.done` ?

Reply:
not sure I follow

Reply:
`isViewAllowed `is now calculated outside, before calling this method